### ISSUE TITLE
7$: E003 false positive for ++a[0]

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#307]
quick-lint-js erroneously reports E003 for the following code:
```js
const a = [42];
++a[0];
```

**Comment by singalhimanshu** [comment]
I claim this for-hire task. I expect payment after I complete this task. I will email the quick-lint-js team if I am assigned this task.
